src/Components/Web.Extensions/src/ProtectedBrowserStorage/ProtectedBrowserStorage.cs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Runtime.InteropServices;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.JSInterop;
+
+namespace Microsoft.AspNetCore.Components.Web.Extensions
+{
+    /// <summary>
+    /// Provides mechanisms for storing and retrieving data in the browser storage.
+    /// </summary>
+    public abstract class ProtectedBrowserStorage
+    {
+        private const string JsFunctionsPrefix = "protectedBrowserStorage";
+
+        private readonly string _storeName;
+        private readonly IJSRuntime _jsRuntime;
+        private readonly IDataProtectionProvider _dataProtectionProvider;
+        private readonly ConcurrentDictionary<string, IDataProtector> _cachedDataProtectorsByPurpose
+            = new ConcurrentDictionary<string, IDataProtector>();
+
+        /// <summary>
+        /// Constructs an instance of <see cref="ProtectedBrowserStorage"/>.
+        /// </summary>
+        /// <param name="storeName">The name of the store in which the data should be stored.</param>
+        /// <param name="jsRuntime">The <see cref="IJSRuntime"/>.</param>
+        /// <param name="dataProtectionProvider">The <see cref="IDataProtectionProvider"/>.</param>
+        protected ProtectedBrowserStorage(string storeName, IJSRuntime jsRuntime, IDataProtectionProvider dataProtectionProvider)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Browser))
+            {
+                throw new PlatformNotSupportedException($"{GetType()} cannot be used when running in Browser WebAssembly.");
```

Comment:
```suggestion
                throw new PlatformNotSupportedException($"{GetType()} cannot be used when running in a browser.");
```

... just because `Browser WebAssembly` sounds so much like `Blazor WebAssembly`, it sounds as if it's some kind of alternate brand name. And the platform name is just "browser".